METRON-1619: Stellar empty collections should be considered false in boolean expressions

cestella opened the pull request:
## Contributor Comments
Similar to #1063 we should consider empty collections and maps as falsey.  This PR builds on #1063, so it should be merged after.

You can test this in the REPL similar to #1063 except substituting `[]` for `is_alert`.


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?


### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

ottobackwards commented on metron-stellar/stellar-common/README.md:
```diff
@@ -54,6 +54,12 @@ The Stellar language supports the following:
 
+### Boolean Expressions
+
+Similar to python and javascript, empty collections (e.g. `[]` and
```
Missing variables vs. NULL variables... this may be confusing.  What this is saying is
we support
- boolean with true or false
- variables that are present but explicitly null or empty
- variables that are NOT present

ottobackwards replied:
This should be in the first pr

ottobackwards replied:
Or am i miss understanding the explicit null case?

cestella replied:
Yeah, what do people think about merging the two PRs into one and reviewing them together?  I split them up anticipating that the community might want one or the other but not necessarily both.  I am thinking that I close #1063 and keep this one.

Thoughts?

cestella commented:
Ok, @simonellistonball I've updated the docs appropriately and @ottobackwards I added tests for match as well as cleaning up the documentation.  This PR should be interpreted to contain both empty list and missing variable falsey.

cestella commented:
@merrimanr would you mind taking a look at this one since you looked at the last one that subsumes.

ottobackwards commented:
+1 by inspection
